### PR TITLE
fix(organizer): movie extras show correct path in job history

### PR DESCRIPTION
**Pull request by Jsakkos**
## Summary

- Movie extra titles (e.g. bonus features on an Apocalypse Now disc) were all showing the **main feature's filename** in the job history detail panel instead of their actual `Extras/Extra N.mkv` paths.
- `organize_movie()` already moved extras correctly on disk but didn't return a source→destination mapping, so `job_manager.py` couldn't correlate which `DiscTitle` mapped to which extra — it fell back to stamping every title with `main_file`.

## What changed

**`backend/app/core/organizer.py`**
- `organize_movie()` now returns two additional keys:
  - `main_source` — basename of the original main file (e.g. `"t00.mkv"`)
  - `extras_mapping` — `{original_basename: dest_Path}` for each extra moved
- All failure/early-return paths include these keys with `None`/`{}` for consistent shape.

**`backend/app/services/job_manager.py`**
- The post-organize title loop now looks up each title's `output_filename` basename in `extras_mapping`.
- Extras get their correct `Extras/Extra N.mkv` `organized_to` path and `is_extra = True`; the main feature continues to get `main_file` as before.

**`backend/tests/pipeline/test_organization_paths.py`**
- Added `TestMovieExtrasMapping` with three tests verifying the new mapping fields (main_source, extras_mapping keys, sequential Extra N naming).

## Reviewer notes

- The bug affected only the history display (`organized_to` field on `DiscTitle`) — files on disk were always organized correctly.
- `is_extra` was already correctly set during the identification phase for most flows; the `t.is_extra = True` assignment in the loop is an extra safety net for the post-rip single-file flow.
- The `else` branch (falling back to `main_file`) handles the edge case where `output_filename` is `None` or doesn't match any entry in the mapping.

## Test plan

- [ ] Run `uv run pytest tests/pipeline/test_organization_paths.py::TestMovieExtrasMapping -v` (requires initialized DB)
- [ ] Run `uv run pytest tests/pipeline/ -v` for pipeline regression check
- [ ] Optionally: simulate a movie disc with extras in DEBUG mode and confirm history detail shows distinct `Extras/Extra N.mkv` paths per title

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: backend/tests/pipeline/test_organization_paths.py [hunk 1]
```diff
@@ -141,6 +141,63 @@ def test_picard_extras_path(self, tmp_path):
...
+        library = tmp_path / "movies"
+        result = organize_movie(staging, "The Godfather", year=1972, library_path=library)
+
+        assert result["success"]
+        dest_names = {dest.name for dest in result["extras_mapping"].values()}
+        assert dest_names == {"Extra 1.mkv", "Extra 2.mkv"}
+
+
+@pytest.mark.pipeline
+class TestTVExtrasOrganizationPaths2:
```
[the remark this answers is not in the document]
Comment (Jsakkos) [reply backend/tests/pipeline/test_organization_paths.py]: Fixed. Moved `test_arrested_dev_extras_path` back into `TestTVExtrasOrganizationPaths` and removed `TestTVExtrasOrganizationPaths2` entirely.

File: backend/app/core/organizer.py [hunk 2]
```diff
@@ -218,7 +218,9 @@ def organize_movie(
         return {
             "success": False,
             "main_file": None,
+            "main_source": None,
```
[the remark this answers is not in the document]
Comment (Jsakkos) [reply backend/app/core/organizer.py]: Fixed. Updated the Returns docstring to `dict with 'success', 'main_file', 'extras', 'extras_mapping', 'error' keys`.

File: backend/app/services/job_manager.py [hunk 3]
```diff
@@ -1623,7 +1625,14 @@ async def _filesystem_progress_monitor():
...
                                 t.organized_from = t.output_filename
-                                t.organized_to = str(organize_result.get("main_file", ""))
+                                output_basename = (
+                                    Path(t.output_filename).name if t.output_filename else None
+                                )
+                                if output_basename and output_basename in extras_mapping:
+                                    t.organized_to = str(extras_mapping[output_basename])
+                                    t.is_extra = True
+                                else:
+                                    t.organized_to = str(organize_result.get("main_file", ""))
```
[the remark this answers is not in the document]
Comment (Jsakkos) [reply backend/app/services/job_manager.py]: Fixed with `or ""` guard. Also dropped `main_source` from the return entirely (YAGNI — no production caller uses it), which removes the parallel None risk on that key too.
